engineering-toolkit/eslint-config/src/typescript.js

```diff
@@ -63,7 +63,7 @@ export function typescript(config, ...overrides) {
         "@typescript-eslint/no-empty-interface": "off",
         "@typescript-eslint/no-explicit-any": "warn",
         "@typescript-eslint/no-inferrable-types": "warn",
-        "@typescript-eslint/no-magic-numbers": "warn",
+        "@typescript-eslint/no-magic-numbers": "off",
```

Comment:
please test this on all gitignored or autogenerated files, because some rules throws only in unified-demo repo

Reply:
Thanks for pointing that out: I've disabled `@typescript-eslint/no-empty-object-type` 9e27d830e99bb0a42c96580335b96f2eb9c57f19

With this configuration I've been able to come down to 15 warnings in total in next/nuxt/middleware/playwright apps Out of those 15 warnings 10 is `@typescript-eslint/no-explicit-any` and 4 are `vue/no-v-html`. So I'd say these changes are enough 😄 

On the side not - I think `@shared/unified-api-eslint-config` from `unified-storefronts` shouldn't be used anymore. Now it's a dependency only in unified-api-sfcc, for every other package we use biome.

Reply:
I do not like that we remove this rule, it helps to keep code clean and readable. What are the arguments in favor of removing this rule besides that our code does not align with it? (in what cases?). In fact, this is one of the first rules about clean code you can possibly learn, no magic number, no strings without variables and so on.

Reply:
No magic number eslint rule is unfortunately not implemented well. 99% of the time it shows a false-positives (Especially with react hooks/vue reactivity). Currently in unified-storefronts is shows more than a hundred of warnings (100% of those are false positives). Here some examples from unified-storefront codebase:

Middleware:
<img width="708" alt="image" src="https://github.com/user-attachments/assets/b86aaa73-f22f-4481-857e-e27755c34488">
<img width="580" alt="image" src="https://github.com/user-attachments/assets/a36718c3-f609-4dc3-97ab-7fcf053de72d">
<img width="474" alt="image" src="https://github.com/user-attachments/assets/43661c5c-1277-4cef-84f9-154798c5117d">

Next.js:
<img width="285" alt="image" src="https://github.com/user-attachments/assets/a92dad5f-9a92-4ca7-8941-d522b1b92655">
<img width="434" alt="image" src="https://github.com/user-attachments/assets/7278892b-e0ec-4d1d-98ce-68c7205454d0">
<img width="467" alt="image" src="https://github.com/user-attachments/assets/30da26c5-2791-4430-bcff-e4defdc8d6ba">
<img width="436" alt="image" src="https://github.com/user-attachments/assets/debf6fdb-d1be-43ae-8d84-2ab6ad7b21aa">
<img width="660" alt="image" src="https://github.com/user-attachments/assets/d6ff155d-5417-4cd4-a00a-c73559ec16a7">

Reply:
All of those I believe to be self-explanatory, but eslint is printing a waning about every single one.

Though, if you believe this rule is important for your team, we can always turn this rule off directly in the storefronts/modules repositories. What do you think @bartoszherba ?

Reply:
In fact all these examples can be reduced to (minValue is just an example):
```
const MIN_QUANTITY = 1;
minValue = MIN_QUANTITY;
```

- Enhances code readability and maintainability.
- Provides a single point of change if the value ever needs to be updated.
- Helps document the intent of the 1.

But all of them seems like no-magic-number false because of the context. I agree it is a "philosophical" area in such cases. I also think that if we remove the rule we would need to be super aware of it on CR.

Reply:
that's fair. I personally prefer to look out for those during the review, because it's not that important matter that eslint needs to notify me of it every time and as you said - it's kind of philosophical and imo arguable.

So, I think we can agree that this rule should be moved directly to the storefronts repo as it's a team decision to ignore it - it shouldn't affect whole comapny